docs/agent/hash.md

```diff
@@ -0,0 +1,84 @@
+## Hash selection
+
+This documentation is intended for Agent developers.
+
+### Collisions
+
+Besides performance, an important consideration during the selection of a hash function is
+how often it creates collisions: a collision in a function is when two different
+inputs give the same output.
+
+Datadog stores samples of the Agent aggregator in a map. In order to have smaller keys
+while storing the values in this map, Datadog uses a hash of some identifying parts
+of the sample: for instance, the metric name and its tags.
+Thus, the map is used as a hashmap.
+
+The Java HashMap implementation contains a "collision resolution" algorithm: when
```

Comment:
Go's hashmap is implemented that way as well :) (or in any language really)

Reply:
Yup true, I've been with this Java one because it explicitly has Hash in its name :D I'm moving it to a google doc I'll adapt there.